pkg/master/reconcilers/endpointsadapter.go

```diff
@@ -103,8 +97,7 @@ func (adapter *EndpointsAdapter) ensureEndpointSliceFromEndpoints(namespace stri
 func endpointSliceFromEndpoints(endpoints *corev1.Endpoints) *discovery.EndpointSlice {
 	endpointSlice := &discovery.EndpointSlice{}
 	endpointSlice.Name = endpoints.Name
-	endpointSlice.Labels = map[string]string{serviceNameLabel: endpoints.Name}
-	endpointSlice.OwnerReferences = []metav1.OwnerReference{{Kind: "Service", Name: endpoints.Name}}
```

Comment:
I'm missing context for the original design that called for setting ownerreferences. Can @thockin or @bowei weigh in on whether this makes sense (here and in all the places that switched)?

If the garbage collector is no longer removing endpoint slices when a service is deleted, is there a controller loop that makes that happen?

Reply:
There were two reasons for omitting owner references here. First, to properly include them would require significant restructuring of code in this package that I was hoping to avoid in this package. The corresponding service may or may not exist at this point as both the service and endpoint(slice) are managed independently here. There's just a simple loop that ensures that both the service and endpoint(slice) resources exist to reference the apiserver. 

Second, the select few resources managed by apiserver like this were never meant to be garbage collected. Since the apiserver is constantly trying to keep these resources in existence, it would actually potentially create more work to have an owner refererence here. If someone were to delete the `kubernetes` service this creates, and the corresponding endpointslice was deleted, it would result in unnecessary cycles for kube-proxy.

I don't know, open to ideas here, this is definitely a strange use case. I was really trying to minimize the scale of changes to this package, but open to another approach.

Reply:
I should add that the reason the owner reference was originally included here is simply that `endpointslicecache` in kube-proxy derived the service name from owner references instead of from the service name label. This PR switches that to derive the service name from the label instead, an approach that seems simpler and cleaner.

Reply:
The only downside for not setting ownerRef here is that when the `default/kubernetes` service is deleted. Its Endpointslice will be not be removed. I think this downside is acceptable compared to refactoring of existing battle proven code since `default/kubernetes` service should always exists during the cluster's life cycle.

Reply:
I view this situation as the case where a user is manually creating an Endpoints object rather than the Endpoints(Slice)Controller (user in this case being the API server). Given that, it is up to the creator of the EndpointSlice object to managed its lifetime. ownerRef is used by the EndpointPointSlice controller to manage lifetimes -- those slices not created by the controller will have to manage lifetimes in their own way.

Also, pragmatically, these objects are basically ensured to always exist while the API server is up, so the life cycle is somewhat moot.

Reply:
> when the default/kubernetes service is deleted. Its Endpointslice will be not be removed

if that happens, when it gets recreated, does the controller autocorrect any extant EndpointSlice objects as needed with the new endpoint data?

Reply:
Yep, the endpoint reconcilers are responsible for this and run consistently to ensure that endpoints (and optionally endpoint slices) exist for this. They are not dependent on any kind of watch/change event for the reconcile loop to run. Start sets up an [async runner](https://github.com/kubernetes/kubernetes/blob/master/pkg/master/controller.go#L170), which includes [RunKubernetesService](https://github.com/kubernetes/kubernetes/blob/master/pkg/master/controller.go#L211),  which loops indefinitely in a `NonSlidingUntil` to call [UpdateKubernetesService](https://github.com/kubernetes/kubernetes/blob/master/pkg/master/controller.go#L230) which ensures the existence of the corresponding service first, followed by endpoints (and endpoint slices if the feature gate is enabled). There's a bit more to it than that, but the gist of it is that this sync logic will run as long as apiserver does, and it does not depend on a change to a specific resource to trigger. This is likely not an ideal flow, but it feels out of scope to change it here.